lib/fog/aws/models/storage/files.rb

```diff
@@ -116,6 +125,11 @@ def new(attributes = {})
         def normalize_headers(data)
           data.headers['Last-Modified'] = Time.parse(data.get_header('Last-Modified'))
           data.headers['ETag'] = data.get_header('ETag').gsub('"','')
+
+          DASHED_HEADERS.each do |header|
+            value = data.get_header(header)
+            data.headers[header] = value if value
```

Comment:
Although I suspect it doesn't cause any problems at present, I'm a little worried about how this may result in data with two different capitalization of the same key in it. I could see how this might lead, for instance, to people then updating the wrong one and being rather confused. I suspect it may be a bit cleaner/safer if we tweak this so that if/when we remap things we also delete the existing value (though maybe this is awkward if we don't know the exact capitalization). Does that make sense? What do you think?

Reply:
@eemus I updated this pull request to clean up the headers. I was surprised that Excon didn't do this for us.

Reply:
Data headers still returning case sensitive was an intentional (albeit, perhaps overly cautious) choice. The idea being that in some cases an end user might specifically care about or check things that relate to case-sensitivity, so we wanted to give back what we got, rather than a mutated version. So, we were trying to provide a raw view, but also make it easier to work with. Again, maybe overthinking it, but there was a reason.